lib/internal/test_runner/test.js

```diff
@@ -605,9 +605,23 @@ class Test extends AsyncResource {
           throw new ERR_INVALID_ARG_TYPE('options.concurrency', ['boolean', 'number'], concurrency);
     }
 
+    // Take timeout from cli
+    const cliTimeout = this.config.timeout;
+    if (cliTimeout != null && cliTimeout !== Infinity) {
+      if (!(this.name === '<root>' && this.parent === null) &&
+      this.constructor.name !== 'FileTest' &&
+      this.constructor.name !== 'Suite') {
+        validateNumber(cliTimeout, 'options.timeout', 0, TIMEOUT_MAX);
+        this.timeout = cliTimeout;
+      }
+    }
+
+    // Take timeout from options
     if (timeout != null && timeout !== Infinity) {
```

Comment:
I've attempted to move this part into the else (when parent !== null) but somehow it caused the `timeout` in `beforeEach` not being set, if we need to move this into else branch above, I can look into this bit of refactoring in a separate PR if that's ok @cjihrig